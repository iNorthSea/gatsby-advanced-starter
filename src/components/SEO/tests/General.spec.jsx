import GeneralTags from "../General";
import { tagListHasEmptyValues, sampleData } from "./TestUtils";

describe("seo module GeneralTags", () => {
  it("generates correct tags on website pages", () => {
    expect.assertions(1);

    const generatedTags = GeneralTags({
      seoData: sampleData.seoWebsite,
      websiteData: sampleData.website,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags on article pages", () => {
    expect.assertions(1);

    const generatedTags = GeneralTags({
      seoData: sampleData.seoArticle,
      websiteData: sampleData.website,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("doesn't generate empty tags", () => {
    expect.assertions(2);

    // Test for website pages
    const generatedWebsiteTags = GeneralTags({
      seoData: sampleData.seoWebsite,
      websiteData: sampleData.website,
    });

    expect(tagListHasEmptyValues(generatedWebsiteTags)).toBe(false);

    // Test for article pages
    const generatedArticleTags = GeneralTags({
      seoData: sampleData.seoArticle,
      websiteData: sampleData.website,
    });

    expect(tagListHasEmptyValues(generatedArticleTags)).toBe(false);
  });
});
