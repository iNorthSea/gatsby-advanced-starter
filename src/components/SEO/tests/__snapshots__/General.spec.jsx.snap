// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seo module GeneralTags generates correct tags on article pages 1`] = `
Array [
  <title>
    Post title
  </title>,
  <meta
    content="Post description."
    name="description"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="image"
  />,
  <link
    rel="shortcut icon"
  />,
]
`;

exports[`seo module GeneralTags generates correct tags on website pages 1`] = `
Array [
  <title>
    Website title
  </title>,
  <meta
    content="Website description."
    name="description"
  />,
  <meta
    content="/logos/logo-1024.png"
    name="image"
  />,
  <link
    rel="shortcut icon"
  />,
]
`;
